x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/detection_rules_client/methods/import_rule.ts

```diff
@@ -6,78 +6,70 @@
+import type { ImportRuleArgs } from '../detection_rules_client_interface';
+import { applyRuleUpdate } from '../mergers/apply_rule_update';
+import { validateMlAuth } from '../utils';
+import { createRule } from './create_rule';
+import { getRuleByRuleId } from './get_rule_by_rule_id';
 
-import { readRules } from '../read_rules';
+interface ImportRuleOptions {
+  rulesClient: RulesClient;
+  prebuiltRuleAssetClient: IPrebuiltRuleAssetsClient;
+  importRulePayload: ImportRuleArgs;
+  mlAuthz: MlAuthz;
+}
 
-export const importRule = async (
-  rulesClient: RulesClient,
-  importRulePayload: ImportRuleArgs,
-  mlAuthz: MlAuthz
-): Promise<RuleResponse> => {
+export const importRule = async ({
+  rulesClient,
+  importRulePayload,
+  prebuiltRuleAssetClient,
+  mlAuthz,
+}: ImportRuleOptions): Promise<RuleResponse> => {
   const { ruleToImport, overwriteRules, allowMissingConnectorSecrets } = importRulePayload;
 
   await validateMlAuth(mlAuthz, ruleToImport.type);
 
-  const existingRule = await readRules({
+  const existingRule = await getRuleByRuleId({
     rulesClient,
     ruleId: ruleToImport.rule_id,
-    id: undefined,
   });
 
   if (existingRule && !overwriteRules) {
     throw createBulkErrorObject({
-      ruleId: existingRule.params.ruleId,
+      ruleId: existingRule.rule_id,
       statusCode: 409,
-      message: `rule_id: "${existingRule.params.ruleId}" already exists`,
+      message: `rule_id: "${existingRule.rule_id}" already exists`,
     });
   }
 
-  let importedInternalRule: RuleAlertType;
-
   if (existingRule && overwriteRules) {
-    const ruleUpdateParams = convertUpdateAPIToInternalSchema({
+    const ruleWithUpdates = await applyRuleUpdate({
+      prebuiltRuleAssetClient,
       existingRule,
       ruleUpdate: ruleToImport,
     });
 
-    importedInternalRule = await rulesClient.update({
+    const updatedRule = await rulesClient.update({
       id: existingRule.id,
-      data: ruleUpdateParams,
-    });
-  } else {
-    /* Rule does not exist, so we'll create it */
-    const ruleCreateParams = convertCreateAPIToInternalSchema(ruleToImport, {
-      immutable: false,
-    });
-
-    importedInternalRule = await rulesClient.create<RuleParams>({
-      data: ruleCreateParams,
-      allowMissingConnectorSecrets,
+      data: convertRuleResponseToAlertingRule(ruleWithUpdates),
     });
+    return convertAlertingRuleToRuleResponse(updatedRule);
   }
 
-  /* Trying to convert an internal rule to a RuleResponse object */
-  const parseResult = RuleResponse.safeParse(internalRuleToAPIResponse(importedInternalRule));
-
-  if (!parseResult.success) {
-    throw new RuleResponseValidationError({
-      message: stringifyZodError(parseResult.error),
-      ruleId: importedInternalRule.params.ruleId,
-    });
-  }
-
-  return parseResult.data;
+  /* Rule does not exist, so we'll create it */
+  return createRule({
+    rulesClient,
+    mlAuthz,
+    rule: {
+      ...ruleToImport,
+      immutable: false,
```

Comment:
`immutable` shouldn't be necessary here, since `ruleToImport` is of type `RuleToImport` where `immutable` is required as `false` (for now)